Go 1.16: TestNocmpIntegration fails

On Fedora Rawhide with Go 1.16 beta 1, TestNocmpIntegration fails with:

```
      testing: go.uber.org/atomic
go.uber.org/atomic
--- FAIL: TestNocmpIntegration (0.00s)
    nocmp_test.go:163: 
        	Error Trace:	nocmp_test.go:163
        	Error:      	"missing $GOPATH\n" does not contain "struct containing nocmp cannot be compared"
        	Test:       	TestNocmpIntegration
FAIL
exit status 1
FAIL	go.uber.org/atomic	0.018s
```

I suspect default gopath was removed in 1.16 thus you need to add it to the environment variables.  I've used the following patch successfully but I'm not sure this is the correct way:

```
From 742e71bb16a909acd89bc9c8178ed1a14d09b9c3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Robert-Andr=C3=A9=20Mauchin?= <zebob.m@gmail.com>
Date: Mon, 25 Jan 2021 04:13:53 +0100
Subject: [PATCH 1/1] Add missing GOPATH environment variable
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Robert-André Mauchin <zebob.m@gmail.com>
---
 nocmp_test.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/nocmp_test.go b/nocmp_test.go
index d4be166..2156350 100644
--- a/nocmp_test.go
+++ b/nocmp_test.go
@@ -155,6 +155,8 @@ func TestNocmpIntegration(t *testing.T) {
 	// Forget OS build enviroment and set up a minimal one for "go build"
 	// to run.
 	cmd.Env = []string{
+		"GO111MODULE=off",
+		"GOPATH=" + filepath.Join(tempdir, "src"),
 		"GOCACHE=" + filepath.Join(tempdir, "gocache"),
 	}
 	cmd.Stderr = &stderr
-- 
2.29.2
```

--- a/nocmp_test.go
+++ b/nocmp_test.go
@@ -114,6 +114,9 @@ func TestNocmpCopy(t *testing.T) {
 	})
 }
 
+// Fake go.mod with no dependencies.
+const _exampleGoMod = `module example.com/nocmp`
+
 const _badFile = `package atomic
 
 import "fmt"
@@ -135,26 +138,29 @@ func TestNocmpIntegration(t *testing.T) {
 	require.NoError(t, err, "unable to set up temporary directory")
 	defer os.RemoveAll(tempdir)
 
-	src := filepath.Join(tempdir, "src")
-	require.NoError(t, os.Mkdir(src, 0755), "unable to make source directory")
-
 	nocmp, err := ioutil.ReadFile("nocmp.go")
 	require.NoError(t, err, "unable to read nocmp.go")
 
 	require.NoError(t,
-		ioutil.WriteFile(filepath.Join(src, "nocmp.go"), nocmp, 0644),
+		ioutil.WriteFile(filepath.Join(tempdir, "go.mod"), []byte(_exampleGoMod), 0644),
+		"unable to write go.mod")
+
+	require.NoError(t,
+		ioutil.WriteFile(filepath.Join(tempdir, "nocmp.go"), nocmp, 0644),
 		"unable to write nocmp.go")
 
 	require.NoError(t,
-		ioutil.WriteFile(filepath.Join(src, "bad.go"), []byte(_badFile), 0644),
+		ioutil.WriteFile(filepath.Join(tempdir, "bad.go"), []byte(_badFile), 0644),
 		"unable to write bad.go")
 
 	var stderr bytes.Buffer
 	cmd := exec.Command("go", "build")
-	cmd.Dir = src
+	cmd.Dir = tempdir
 	// Forget OS build enviroment and set up a minimal one for "go build"
-	// to run.
+	// to run. We need GOPATH and GOCACHE set for the compiler to run but
+	// we don't do anything with them.
 	cmd.Env = []string{
+		"GOPATH=" + filepath.Join(tempdir, "gopath"),
 		"GOCACHE=" + filepath.Join(tempdir, "gocache"),
 	}
 	cmd.Stderr = &stderr

